bullet/src/SDFFeatures.cc

```diff
@@ -332,32 +276,38 @@ Identity SDFFeatures::ConstructSdfJoint(
   math::Vector3d pivotParent, pivotChild, axisParent, axisChild;
   math::Pose3d pose;
 
-  pivotParent = (_sdfJoint.RawPose() + this->links.at(childId)->pose).Pos();
-  pivotChild = (_sdfJoint.RawPose() + this->links.at(childId)->pose).Pos();
-
-  pose = this->links.at(parentId)->pose;
-
-  pivotParent -= pose.Pos();
-
-  pivotParent = pose.Rot().RotateVectorReverse(pivotParent);
-  axisParent = pose.Rot().RotateVectorReverse(axis);
-  axisParent = axisParent.Normalize();
+  if (parentId != worldId) {
+    pivotParent = (_sdfJoint.RawPose() + this->links.at(childId)->pose).Pos();
+    pose = this->links.at(parentId)->pose;
+    pivotParent -= pose.Pos();
+    pivotParent = pose.Rot().RotateVectorReverse(pivotParent);
+    axisParent = pose.Rot().RotateVectorReverse(axis);
+    axisParent = axisParent.Normalize();
+  }
 
+  pivotChild = (_sdfJoint.RawPose() + this->links.at(childId)->pose).Pos();
   pose = this->links.at(childId)->pose;
-
   pivotChild -= pose.Pos();
-
   pivotChild = pose.Rot().RotateVectorReverse(pivotChild);
   axisChild = pose.Rot().RotateVectorReverse(axis);
   axisChild = axisChild.Normalize();
 
-  btHingeConstraint* joint = new btHingeConstraint(
-    *this->links.at(childId)->link,
-    *this->links.at(parentId)->link,
-    convertVec(ignition::math::eigen3::convert(pivotChild)),
-    convertVec(ignition::math::eigen3::convert(pivotParent)),
-    convertVec(ignition::math::eigen3::convert(axisChild)),
-    convertVec(ignition::math::eigen3::convert(axisParent)));
+  btHingeAccumulatedAngleConstraint* joint;
+  if (parentId != worldId) {
+    joint = new btHingeAccumulatedAngleConstraint(
```

Comment:
Do this really need to be created as a `btHingeAccumulatedAngleConstraint` or can it be created as a `btHingeConstraint` and then casted?

Reply:
I didn't try setting it to btHingeConstraint, as we are using getHingeAccumulatedAngle in the position getters. Is it there a good reason I may be missing to create it as btHingeConstraint?

Reply:
It was a `btHingeConstraint* joint = new btHingeConstraint(` before you changed it, I'm asking the reason behind the change

Reply:
Sure. In the documentation `getHingeAccumulatedAngle` is only available for `btHingeAccumulatedConstraint`, it's weird the way it was before, and probably the angle returned wasn't correct.
 https://pybullet.org/Bullet/BulletFull/classbtHingeAccumulatedAngleConstraint.html#a88d009b1a4c8ba6ad95790358c5279bf

Reply:
Ok, makes sense